### PR TITLE
[FLOC-4432] Add Pure Storage backend driver documentation and release notes

### PR DESCRIPTION
**Pull request by wallrj**
Fixes: https://clusterhq.atlassian.net/browse/FLOC-4432

See:
 * http://clusterhq-staging-docs.s3-website-us-east-1.amazonaws.com/purestorage-FLOC-4432/releasenotes/index.html#next-release
 * http://clusterhq-staging-docs.s3-website-us-east-1.amazonaws.com/purestorage-FLOC-4432/supported/index.html#hardware-storage-devices
 * http://clusterhq-staging-docs.s3-website-us-east-1.amazonaws.com/purestorage-FLOC-4432/flocker-features/pure-storage-configuration.html#pure-storage-backend
 * http://clusterhq-staging-docs.s3-website-us-east-1.amazonaws.com/purestorage-FLOC-4432/flocker-features/storage-backends.html#community-supported-drivers

### CODE REVIEW

File: docs/flocker-features/pure-storage-configuration.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+=================================
+Pure Storage Block Device Backend
+=================================
+
+.. begin-body
+
+Pure Storage provides a Flocker backend plugin.
+
+For more information, including installation, testing and usage instructions, visit the `Pure Storage Flocker Driver on GitHub <https://github.com/sdodsley/purestorage-flocker-driver>`_
```
Comment (myechuri) [inline docs/flocker-features/pure-storage-configuration.rst]: @wallrj : this is not the final location of the driver. Will let you know once it is available (early next week).

Comment (patrick-east) [inline docs/flocker-features/pure-storage-configuration.rst]: Hey, if you want to update the URL, the official repo is at https://github.com/PureStorage-OpenConnect/purestorage-flocker-driver

Its private right now, but as soon as I've got the go-ahead I'll switch it to a public repo.

Comment (wallrj) [inline docs/flocker-features/pure-storage-configuration.rst]: Thanks @patrick-east,
I updated the link.
Will you update the Flocker documentation link on your repo:

> Install ClusterHQ/Flocker
> https://docs.clusterhq.com/en/0.3.2/gettingstarted/index.html

Should link to the latest stable version of Flocker, which is currently:
 * https://docs.clusterhq.com/en/1.12.0/

And this week we're hoping to release 1.13.0 so it'd be great if you could test the driver against Flocker master builds:
 * http://build.clusterhq.com/results/omnibus/master/

Comment (patrick-east) [inline docs/flocker-features/pure-storage-configuration.rst]: I ninja edited the README, its just pointing to the latest docs page so the link wont go stale.

I'll upgrade one of my test envs to 1.13 and make sure things are still ok. Were there any major changes I should check for specifically?

Comment (myechuri) [inline docs/flocker-features/pure-storage-configuration.rst]: @patrick-east : i think you did test with the latest Flocker bits - the tests that ran on your system are from top of tree, and were not present in 0.3.2.

Comment (patrick-east) [inline docs/flocker-features/pure-storage-configuration.rst]: We tested with 1.12.0, and with master of the Flocker tree as of a few days ago. I think the 0.3.2 was a remnant of some old doc links we had.

File: docs/flocker-features/pure-storage-configuration.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.. _pure-storage-backend:
+
+=================================
+Pure Storage Block Device Backend
+=================================
+
+.. begin-body
+
+Pure Storage provides a Flocker backend plugin.
```
Comment (ferrantim) [inline docs/flocker-features/pure-storage-configuration.rst]: change to "Pure Storage provides a Flocker backend plugin for FLASHARRAY//m All flash storage system."

Then add a setence at says "FlashArray//m enables you to transform your rack, data center, or cloud, with an afordable all-flash array capable of consolidating all your key business applications."

**Comment by ferrantim** [comment]
Looks good richard.  Just one comment.

**Pull request merged**